### PR TITLE
feat: add Y combinator

### LINKED ISSUE #130: Y Combinator
http://kestas.kuliukas.com/YCombinatorExplained/

### PR DESCRIPTION
**Pull request by char0n**
Closes #130

### CODE REVIEW

**Comment by char0n** [comment]
@rockymadden review PLS

File: src/Y.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+ *
+ * The Y combinator is an interesting function which only works with functional languages,
+ * showing how recursion can still be done even without any variable or function declarations,
+ * only functions and parameters
+ *
+ * @func Y
+ * @memberOf RA
+ * @since {@link https://char0n.github.io/ramda-adjunct/2.3.0|v2.3.0}
+ * @category Function
+ * @sig (... ->  -> null
```
Comment (char0n) [inline src/Y.js]: Missing valid signature

File: src/index.d.ts [hunk 2]
```diff
@@ -659,6 +659,15 @@ declare namespace RamdaAdjunct {
...
         defaultWhen(predicate: Function): <DefVal, Val>(defaultVal: DefVal) => (val: Val) => DefVal | Val;
 
+        /**
+         * Y-combinator
+         *
+         * The Y combinator is an interesting function which only works with functional languages,
+         * showing how recursion can still be done even without any variable or function declarations,
+         * only functions and parameters
+         */
+        Y(fnMaker: Function): Function;
```
Comment (char0n) [inline src/index.d.ts]: fnMaker -> le

**Comment by char0n** [comment]
@BjornMelgaard thx for CR.

**Comment by rockymadden** [comment]
Seriously cool!

**Pull request merged**